### PR TITLE
feat: Support nested properties and CompositionPropertySet in ExpressionAnimation

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.Composition/Composition/CompositionObject.SubPropertyHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+
+using System;
+using System.Numerics;
+using Windows.UI;
+
+namespace Microsoft.UI.Composition;
+
+partial class CompositionObject
+{
+	private protected Color GetColor(ReadOnlySpan<char> subPropertyName, Color existingValue, object? propertyValue)
```
Comment (dr1rrb) [inline src/Uno.UI.Composition/Composition/CompositionObject.SubPropertyHelpers.cs:11]:
```suggestion
	private protected Color UpdateColor(ReadOnlySpan<char> subPropertyName, Color existingValue, object? propertyValue)
```

Not sure if it's the right name, but for me I would not expect a `Get<Something>` to update only sub properties. (Same for all methods below)

File: src/Uno.UI.Composition/Composition/CompositionPropertySet.cs [hunk 2]
```diff
@@ -80,12 +80,13 @@ private void SetValue<T>(string propertyName, T value)
 		}
 
 		// TODO
```
Comment (dr1rrb) [inline src/Uno.UI.Composition/Composition/CompositionPropertySet.cs:82]: Still `//TODO` ? If not, why not just remove the override?

File: src/Uno.UI.Composition/Composition/CompositionPropertySet.cs [hunk 3]
```diff
@@ -80,12 +80,13 @@ private void SetValue<T>(string propertyName, T value)
...
-		private protected override bool IsAnimatableProperty(string propertyName)
-			=> false;
+		private protected override bool IsAnimatableProperty(ReadOnlySpan<char> propertyName)
+			=> base.IsAnimatableProperty(propertyName);
 
 		// TODO
```
Comment (dr1rrb) [inline src/Uno.UI.Composition/Composition/CompositionPropertySet.cs:86]: Same :)

**Review by dr1rrb — APPROVED** [review]

File: src/Uno.UI.Composition/Composition/CompositionObject.SubPropertyHelpers.cs (hunk 1 above)
Comment (Youssef1313) [reply src/Uno.UI.Composition/Composition/CompositionObject.SubPropertyHelpers.cs]: Fixed.

File: src/Uno.UI.Composition/Composition/CompositionPropertySet.cs (hunk 2 above)
Comment (Youssef1313) [reply src/Uno.UI.Composition/Composition/CompositionPropertySet.cs]: Yup I think we might not need special handling for CompositionPropertySet as the virtual implementation should be doing the right thing. Removed the override. We can adjust if we discovered any needed special handling.

**Pull request merged**